### PR TITLE
Checkpoint 5: Views

### PR DESCRIPTION
**Pull request by CRClothier**
- Implement the design from the sneak peek wireframes.
- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,10 @@
-<h1>A list of all users</h1>
\ No newline at end of file
+<h1>A list of all users</h1>
+<% @users.each do |user| %>
```
Comment (SajjadAhmad14) [inline app/views/users/index.html.erb:2]: Every user on `app/views/index.html.erb`should have a link and clicking on it should take the user to the user's show page.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,22 @@
...
+  <div class="user-info">
+    <h2><%= @user.name%></h2>
+    <p>Number of Posts: <%= @user.posts_counter || 0 %></p>
+  </div>
+</div>
+<div class="bio">
+  <h3>Bio</h3>
+  <p><%= @user.bio %></p>
+</div>
+
```
Comment (SajjadAhmad14) [inline app/views/users/show.html.erb:12]: There should be `See all Post` link that redirects the user to the user's post index.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,5 +1,12 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
```
Comment (SajjadAhmad14) [inline app/controllers/posts_controller.rb:2]: There is an error when the user tries to see the posts index page. Kindly see the screenshot below.
posts

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,5 +1,10 @@
 class UsersController < ApplicationController
-  def index; end
+  def index
+    @users = User.all
+  end
 
-  def show; end
+  def show
```
Comment (SajjadAhmad14) [inline app/controllers/users_controller.rb:6]: The user encounters this error when he tries to go to the user's show page.
show

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Chris, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by CRClothier — COMMENTED** [review]
yes

**Pull request merged**